samples/arm-none-eabi/reported.by

```diff
@@ -0,0 +1,310 @@
+reporter_name="Jasmin Jessich"
+reporter_url="http://jasmin.anw.at/"
+reporter_comment="NOTE: Please read samples/arm-none-eabi/reported.by prior to any built!"
+
+#
+# This configuration will build an arm-none-eabi compiler to be used for
+# embedded development. The original build script can be found here:
+#    https://launchpad.net/gcc-arm-embedded
+# To see the original build script, you have to download the installer:
+#    https://launchpad.net/gcc-arm-embedded/5.0/5-2015-q4-major
+# 
+# I tried to mimic the original configure options and environment settings with
+# crosstool-ng (CT_NG) as close as possible. I really analysed every option
+# which was not provided by CT-NG or which was added. The description in this
+# file explains each diversion from the original.
+# 
+# NOTE:   newlib-nano is currently not built!
+# 
+# The original uses the git/SVN versions of binutils-gdb, gcc and newlib. CT-NG
+# doesn't currently support using a git or SVN repository for the source
```

Comment:
On 03/21/2016 06:34 PM, Jasmin Jessich wrote:
>
> In samples/arm-none-eabi/reported.by 
> <https://github.com/crosstool-ng/crosstool-ng/pull/372#discussion_r56925719>:
>
> > +#
> > +# This configuration will build an arm-none-eabi compiler to be used for
> > +# embedded development. The original build script can be found here:
> > +#    https://launchpad.net/gcc-arm-embedded
> > +# To see the original build script, you have to download the installer:
> > +#    https://launchpad.net/gcc-arm-embedded/5.0/5-2015-q4-major
> > +#
> > +# I tried to mimic the original configure options and environment settings with
> > +# crosstool-ng (CT_NG) as close as possible. I really analysed every option
> > +# which was not provided by CT-NG or which was added. The description in this
> > +# file explains each diversion from the original.
> > +#
> > +# NOTE:   newlib-nano is currently not built!
> > +#
> > +# The original uses the git/SVN versions of binutils-gdb, gcc and newlib. CT-NG
> > +# doesn't currently support using a git or SVN repository for the source
>
>     the question really was, is there anything in the /old/
>     configuration that the /new/ configuration
>     does not provide?
>
> It provides a compiler as close as possible to the original from ARM 
> for embedded development (arm-none-eabi). It provides special settings 
> for newlib and it builds for the embedded processor types (armv?-m).
> Let me ask the question in the other direction. What was the intention 
> for the "arm-unknown-none-eabi" ?
>
I have no idea, which is why I think it would be good to replace the 
existing sample with a configuration that has a known purpose :)
We can ask the original submitter (Yann Morin), he's available sometimes 
at the IRC. I'll flag him about this pull request next time I see him 
online.

> There is no further explanation within the reported.by file to explain 
> it. So it is really hard to guess from the sample parameters itself.
>
>     And, when choosing the name for the sample, make sure that 'ct-ng
>     saveconfig' agrees with
>     your naming :)
>
> I tried this already, but I had no luck. I would like to name the 
> tools exactly like ARM did it for their compiler (arm-none-eabi). But 
> I would like to have the name of the sample "arm-none-eabi-embedded". 
> I don't see how I need to set the options to convince 'ct-ng 
> saveconfig' to do what I like. That's the reason, why I called it 
> simple "arm-none-eabi".
> Any hints?
>
'ct-ng saveconfig' follows pretty strict rules as to naming - 
"[host-tuple,]target-tuple". So, it all boils down to how your 
configuration gets named by CT_DoBuildTargetTuple in scripts/functions. 
Perhaps we can add support for a configurable suffix to sample name.

But I think it is redundant in this case, given that EABI already stands 
for "Embedded ABI" [see https://wiki.debian.org/ArmEabiPort].

All that, I guess, is one more reason to replace existing 
arm-unknown-none-eabi sample with this one.
>
> And I plan another sample for mingw, where the compiler shall be named 
> also "arm-none-eabi", but the sample 
> arm-none-eabi-embedded-w64-mingw32 (or similar).
>
That would be named as something 
'i686-w64-mingw32,arm-unknown-none-eabi' by saveconfig.

Regards,
Alexey.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/crosstool-ng/crosstool-ng/pull/372/files/9ea84294c84ab9ba765e8aeafa3648c36edc1f28#r56925719>
>